### PR TITLE
Show historical OldDot Report Actions in Plain text

### PR DESCRIPTION
**Pull request by eh2077**
### Details

This PR incorporates commits from https://github.com/Expensify/App/pull/37022 and works on them.

### Fixed Issues

$ https://github.com/Expensify/App/issues/36023
$ https://github.com/Expensify/App/issues/34968
$ https://github.com/Expensify/App/issues/34783
$ https://github.com/Expensify/App/issues/32602
PROPOSAL:

#### Pre-testing steps
1. Create a new collect policy on OldDot and;
      i. Enable scheduled submit with a frequency of Instant (this should happen by default with Expensify accounts, or accounts on paidPolicyInstantSubmit beta)
     ii. Make sure the approval mode is "Submit & Approve"
    iii. Invite an additional user to the policy to play the member role in these testing flows
2. Run the following script in the chrome console (in OldDot, when the workspace Settings page is open) to enable the 
    ```
    policy for workspace creation for a paid plan.
    p = Policy.getCurrent();
    p.policy.isPolicyExpenseChatEnabled = "true";
    p.save();
    ```
#### Test steps

1. In OldDot: As submitter, make sure ^ new workspace is your "Active" / default workspace
2. In OldDot: As submitter, create a report and add an expense
3. In OldDot: As Admin, approve the report - then Unapprove the report
4. In NewDot: As both accounts, view the expense report & verify you can see both of these report actions:

     i. <admin> final approved this report
    ii. <admin> unapproved this report

5. Verify that <admin> `unapproved` this report report action appear.

- [x] Verify that no errors appear in the JS console

### QA Steps

Same as tests
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/117511920/e527bdd5-8117-445a-83a1-3869962c1429

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-03-11 at 11 10 26 PM

</details>

### CODE REVIEW

**Comment by eh2077** [comment]
@twisterdotcom Kindly help to review if all oldDot actions needed are included, thanks! You can just comment here if there're other types need to add. I'll help update them then.

**Comment by twisterdotcom** [comment]
@eh2077 - I checked yesterday when chatting this through with @dylanexpensify - the only one missing was the EXPORTEDTOCSV one which I pushed above. Other than that, we're good to go!

File: src/pages/home/report/ReportActionItem.tsx [hunk 1]
```diff
@@ -517,6 +517,40 @@ function ReportActionItem({
...
+                CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENTDELAYED,
+                CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENTREQUESTED,
+                CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENTSETUP,
+                CONST.REPORT.ACTIONS.TYPE.SELECTEDFORRANDOMAUDIT,
+                CONST.REPORT.ACTIONS.TYPE.SHARE,
+                CONST.REPORT.ACTIONS.TYPE.STRIPEPAID,
+                CONST.REPORT.ACTIONS.TYPE.TAKECONTROL,
+                CONST.REPORT.ACTIONS.TYPE.UNAPPROVED,
+                CONST.REPORT.ACTIONS.TYPE.UNSHARE,
+            ].find((oldDotActionName) => oldDotActionName === action.actionName)
```
[the remark this answers is not in the document]
Comment (eh2077) [reply src/pages/home/report/ReportActionItem.tsx:550]: Good question! We'll get type check error if using `includes`

[the remark this answers is not in the document]
Comment (eh2077) [reply src/pages/home/report/ReportActionItem.tsx:550]: > Should we move it into a util? `isOldDotReportAction`

I think it's a good idea. Will update it.

File: src/libs/ReportActionsUtils.ts [hunk 2]
```diff
@@ -808,6 +808,14 @@ function getMemberChangeMessageFragment(reportAction: OnyxEntry<ReportAction>):
     };
 }
 
+/**
+ * MARKEDREIMBURSED reportActions come from marking a report as reimbursed in OldDot. For now, we just
+ * concat all of the text elements of the message to create the full message.
+ */
+function getMarkedReimbursedMessage(reportAction: OnyxEntry<ReportAction>): string {
+    return reportAction?.message?.map((element) => element.text).join('') ?? '';
```
[the remark this answers is not in the document]
Comment (eh2077) [reply src/libs/ReportActionsUtils.ts:816]: Hmm, good point. I just added this method back from where it was deleted https://github.com/Expensify/App/commit/4b7c614eb41009d14921ffc4b0206dc0fd441358#diff-b8041d6cb129cc7b8a6223f8ffa570e79c326d8ea4affbad35c33cf6c8ea660dL807

Maybe we can ask internal engineer to share a sample of oldDot messages to help us determine how to join the text properly.

[the remark this answers is not in the document]
Comment (Beamanator) [reply src/libs/ReportActionsUtils.ts:816]: AFAIK we don't need to add spaces b/c we already add them in the backend - you possibly can already see some of the `message` arrays in Onyx, I don't have an example I can think of off the top of my head but let me know if you want me to get one and i can

**Comment by eh2077** [comment]
@hoangzinh Good catch. Yes, in this test case, the `approved` message is shown in LHN but the `unapproved` msg is not. See 
https://github.com/Expensify/App/blob/4e599b2cd6061b52d3964dc69e7575e210f0581d/src/libs/ReportActionsUtils.ts#L912-L914

@twisterdotcom Do we need to show those system messages from oldDot actions in LHN? If we want to show them, then I think we need to sort out them case by case because many of them may need corresponding Spanish translations. Or I guess we can just omit showing them in LHN because they're legacy actions which won't be added to newDot?

**Comment by twisterdotcom** [comment]
@eh2077 - we can omit them from the LHN for now. We have plans to slowly migrate these all to native newDot actions - this is more a temporary fix to resolve confusion before we build them into newDot with Spanish translations and better wording to fit the chat flow.

**Review by hoangzinh — APPROVED** [review]
LGTM

**Review by Beamanator — APPROVED** [review]
LGTM!

**Comment by Beamanator** [comment]
Uhhh the checklist point failed even though it was filled out here - https://github.com/Expensify/App/pull/38013#issuecomment-1997590935

File: src/CONST.ts [hunk 3]
```diff
@@ -573,28 +573,57 @@ const CONST = {
...
+                CHANGEPOLICY: 'CHANGEPOLICY', // OldDot Action
+                CHANGETYPE: 'CHANGETYPE', // OldDot Action
                 CHRONOSOOOLIST: 'CHRONOSOOOLIST',
                 CLOSED: 'CLOSED',
                 CREATED: 'CREATED',
+                DELEGATESUBMIT: 'DELEGATESUBMIT', // OldDot Action
+                DELETEDACCOUNT: 'DELETEDACCOUNT', // OldDot Action
+                DONATION: 'DONATION', // OldDot Action
+                EXPORTEDTOCSV: 'EXPORTEDTOCSV', // OldDot Action
+                EXPORTEDTOINTEGRATION: 'EXPORTEDTOINTEGRATION', // OldDot Action
```
Comment (arosiclair) [inline src/CONST.ts:592]: Noticed the value for this one is off. It should just be `EXPORTINTEGRATION` ([src](https://github.com/Expensify/Auth/blob/45f33f3dda3aac0172940cb6f0e9353229ec4f1f/auth/lib/Report.cpp#L181)). @Beamanator can you update that? We're going to need it for the QBO doc I'm working on.

Comment (Beamanator) [reply src/CONST.ts:592]: Good catch @arosiclair ! fixed here: https://github.com/Expensify/App/pull/42242

**Pull request merged**